realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java

```diff
@@ -2461,10 +2461,10 @@ public void isNotNull_linkField() {
         // 10 Date
         assertEquals(1, realm.where(NullTypes.class).isNotNull(
                 NullTypes.FIELD_OBJECT_NULL + "." + NullTypes.FIELD_DATE_NULL).count());
+        // FIXME: Currently, Realm Core does not support isNotNull() query for nested link field.
```

Comment:
I don't know.  I just noticed that this code was commented out and added the "FIXME" to make it more visible.  Will check with core...